pulsar-broker/src/test/java/org/apache/pulsar/client/api/RetryTopicTest.java

```diff
@@ -457,4 +462,58 @@ public void testRetryTopicByCustomTopicName() throws Exception {
         checkConsumer.close();
     }
 
+
+    @Test
+    public void testRetryTopicException() throws Exception {
+        final String topic = "persistent://my-property/my-ns/retry-topic";
+        final int maxRedeliveryCount = 2;
+        final int sendMessages = 1;
+        // subscribe before publish
+        Consumer<byte[]> consumer = pulsarClient.newConsumer(Schema.BYTES)
+                .topic(topic)
+                .subscriptionName("my-subscription")
+                .subscriptionType(SubscriptionType.Shared)
+                .enableRetry(true)
+                .receiverQueueSize(100)
+                .deadLetterPolicy(DeadLetterPolicy.builder()
+                        .maxRedeliverCount(maxRedeliveryCount)
+                        .retryLetterTopic("persistent://my-property/my-ns/my-subscription-custom-Retry")
+                        .build())
+                .subscriptionInitialPosition(SubscriptionInitialPosition.Earliest)
+                .subscribe();
+
+        Producer<byte[]> producer = pulsarClient.newProducer(Schema.BYTES)
+                .topic(topic)
+                .create();
+        for (int i = 0; i < sendMessages; i++) {
+            producer.send(String.format("Hello Pulsar [%d]", i).getBytes());
+        }
+        producer.close();
+
+        // mock a retry producer exception when reconsumelater is called
+        MultiTopicsConsumerImpl multiTopicsConsumer = (MultiTopicsConsumerImpl) consumer;
+        List<ConsumerImpl> consumers = multiTopicsConsumer.getConsumers();
+        for (ConsumerImpl c : consumers) {
+            Set<Field> deadLetterPolicyField =
+                    ReflectionUtils.getAllFields(c.getClass(), ReflectionUtils.withName("deadLetterPolicy"));
+
+            if (deadLetterPolicyField.size() != 0) {
+                Field field = deadLetterPolicyField.iterator().next();
+                field.setAccessible(true);
+                DeadLetterPolicy deadLetterPolicy = (DeadLetterPolicy) field.get(c);
+                deadLetterPolicy.setRetryLetterTopic("#persistent://invlaid-topic#");
+            }
+        }
```

Comment:
```suggestion
        MultiTopicsConsumerImpl<byte[]> multiTopicsConsumer = (MultiTopicsConsumerImpl<byte[]>) consumer;
        List<ConsumerImpl<byte[]>> consumers = multiTopicsConsumer.getConsumers();
        for (ConsumerImpl<byte[]> c : consumers) {
            c.getDeadLetterPolicy().setRetryLetterTopic("#persistent://invlaid-topic#");
        }
```

Instead of depending on reflection, I'd prefer to add a `@VisibleForTesting` method `getDeadLetterPolicy` in `ConsumerImpl`. I don't know whether it's the conventions in Pulsar, though.

Reply:
here I only want to construct an exception, @VisibleForTesting also ok ,but it seems  not necessary

Reply:
No need to add @VisibleForTesting for this test.

Reply:
@Technoboy- fair enough. @leizhiyuan You may update the generic type parameter parts though.